cinema/views.py

```diff
@@ -1,45 +1,121 @@
-from rest_framework.decorators import api_view
+from rest_framework import status, generics, mixins, viewsets
 from rest_framework.response import Response
-from rest_framework import status
-
+from rest_framework.views import APIView
 from django.shortcuts import get_object_or_404
+from cinema.models import Movie, Genre, Actor, CinemaHall
+from cinema.serializers import (
+    MovieSerializer,
+    ActorSerializer,
+    GenreSerializer,
+    CinemaHallSerializer,
+)
+
+
+class GenreList(APIView):
+    def get(self, request):
+        movies = Genre.objects.all()
```

Comment:
The variable `movies` should be renamed to `genres` to accurately reflect the data being retrieved from the `Genre` model.